File: peuler/ex12/ex12.py
import math

def nthTriang(n):
    s = 0
    for i in range(1, n+1):
        s += i
    return s

def countDivs(n):
    count = 2
    for i in range(2, int(math.sqrt(n)) + 1):
        if (n % i == 0):
            count += 1
    return count


nth = 100
numberOfDivs = countDivs(nthTriang(nth))

while (numberOfDivs <= 500):
    numberOfDivs = countDivs(nthTriang(nth))
    nth += 1

print(str(nthTriang(nth-1)) + " is the " + str(nth - 1) + " triangular number with over 500 divs.")

File: peuler/ex12/ex12.cpp
#include<iostream>
#include<math.h>

unsigned long int nthTriang(int n) {
  int s = 0;

  if (n == 1) s = 1;
  else s = nthTriang(n-1) + n;

  return s;
}

unsigned long int countDivs(unsigned long int n) {
  int count = 2;

  for (int i = 2; i <= (int) sqrt(n); i++)
	if (n % i == 0) count++;

  return count;
}

int main() {
  unsigned long int numberOfDivs;
  int nth = 100;

  numberOfDivs = countDivs(nthTriang(nth));

  while (numberOfDivs <= 500)
	numberOfDivs = countDivs(nthTriang(nth++));
	

  printf("%lu is the %d triangular number with over 500 divs.\n", nthTriang(nth-1), nth-1);

  return 0;
}
